phys2bids/split2phys.py

```diff
@@ -0,0 +1,179 @@
+    if len(tr_list) != 1 and len(ntp_list) < len(tr_list):
+        raise Exception('Multiple sequence types have been listed in tr,'
+                        'but the number of run is less than types of sequence')
+    # 2 sequence types, 3 runs ; which one is it??????
+    if len(tr_list) != 1 and len(tr_list) < len(ntp_list):
+        raise Exception('Multiple sequence types have been listed in tr,'
+                        'but the number of run doesn\'t match')
+
+    # Check out this page for all the builtin errors:
+    # https://docs.python.org/3/library/exceptions.html#bltin-exceptions
+
+    # if multiple runs of same sequence in recording - pad the list with same value
+    if len(tr_list) == 1:
+        tr_list = tr_list * len(ntp_list)
+
+    # Import right interface to read the file
+    if ftype == 'acq':
+        from phys2bids.interfaces.acq import populate_phys_input
+    elif ftype == 'txt':
+        from phys2bids.interfaces.txt import populate_phys_input
+    else:
+        # #!# We should add a logger here.
+        raise NotImplementedError('Currently unsupported file type.')
+
+    # Actually read file!
+    LGR.info(f'Reading the file {infile}')
+    phys_in = populate_phys_input(infile, chtrig)  # phys_in is a BlueprintInput object
+    LGR.info('Reading infos')
+    phys_in.print_info(filename)
+
+    if chplot != '' or info:
+        viz.plot_all(phys_in.ch_name, phys_in.timeseries, phys_in.units,
+                     phys_in.freq, infile, chplot)
+    # If only info were asked, end here.
+    if info:
+        return
+
+    # Sum of values in ntp_list should be equivalent to num_timepoints_found
+    phys_in.check_trigger_amount(chtrig=chtrig, thr=thr,
+                                 num_timepoints_expected=sum(ntp_list),
+                                 tr=1)  # TODO : define a non-hard-coded value
+
+    # Check that sum(ntp_list) is equivalent to num_timepoints_found, else bye!
+    # num_timepoints_found becomes an attribute of the object when you call check_trigger_amount
+    if phys_in.num_timepoints_found != sum(ntp_list):
+        raise ValueError()  # not sure if it's the good one
+        # TODO : automatize tps correction
+
+    # Initialize dictionaries to save phys_in endpoints
+    run_endpoints = {}
+
+    # initialise start index as 0
+    start_index = 0
+
+    for run_idx, run_tps in enumerate(ntp_list):
+        # ascertain run length and and (re)initialise Blueprint object
+        if run_idx == 0:
+            phys_in.check_trigger_amount(ntp=run_tps, tr=tr_list[run_idx])
+        else:
+            phys_in = phys_in.delete_at_index([start_index:end_index])
+            phys_in.check_trigger_amount(ntp=run_tps, tr=tr_list[run_idx])
+        # define padding - 20s * freq of trigger - padding is in nb of samples
+        padding = 20 * phys_in.freq[chtrig]
+
+        # LET'S START NOT SUPPORTING MULTIFREQ - end_index is nb of samples in run+start+first_trig
```

Comment:
Could you elaborate on why you don't want to support multifreq yet?

Reply:
Sure thing!
I'm doing it in the main conversation though.